src/xp.runner/io/Cygwin.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Collections.Generic;
+using Microsoft.Win32;
+
+namespace Xp.Runners.IO
+{
+    static class Cygwin
+    {
+        private const string CYGDRIVE_PATH = "/cygdrive/";
+        private const string INSTALLATIONS = @"Software\Cygwin\Installations";
+
+        private static byte[] SYMLINK_COOKIE = new byte[] { 33, 60, 115, 121, 109, 108, 105, 110, 107, 62 };
+
+        /// <summary>Determine whether we're runnning inside Cygwin</summary>
+        public static bool Active { get { return null != Environment.GetEnvironmentVariable("SHELL"); } }
+
+        /// <summary>Determine Cygwin installation directories. Caches information</summary>
+        public static IEnumerable<string> Installations()
+        {
+            var installed = Registry.CurrentUser.OpenSubKey(INSTALLATIONS) ?? Registry.LocalMachine.OpenSubKey(INSTALLATIONS);
+            if (null == installed)
```

Comment:
Seems like this should use HKEY_LOCAL_MACHINE\SOFTWARE\Cygwin\setup\rootdir instead...

https://www.cygwin.com/faq.html#faq.setup.registry